src/overlays/actors/ovl_Bg_Mori_Hashira4/z_bg_mori_hashira4.c

```diff
@@ -20,27 +36,143 @@ const ActorInit Bg_Mori_Hashira4_InitVars = {
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 700, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+static Gfx* sDisplayLists[] = { 0x06001300, 0x06008840 };
+
+static s16 sUnkTimer; // seems to be unused
+
+void BgMoriHashira4_SetupAction(BgMoriHashira4* this, BgMoriHashira4ActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgMoriHashira4_InitDynaPoly(BgMoriHashira4* this, GlobalContext* globalCtx, ColHeader* collision, s32 moveFlag) {
+    s32 pad;
+    ColHeader* colHeader;
+    s32 pad2;
+
+    colHeader = NULL;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(collision, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_mori_hashira4.c", 155,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgMoriHashira4_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriHashira4* this = THIS;
+
+    this->switchFlag = (this->dyna.actor.params >> 8) & 0x3F;
+    this->dyna.actor.params &= 0xFF;
+
+    if (this->dyna.actor.params == 0) {
+        BgMoriHashira4_InitDynaPoly(this, globalCtx, &D_06001AF8, DPM_UNK3);
+    } else {
+        BgMoriHashira4_InitDynaPoly(this, globalCtx, &D_060089E0, DPM_UNK);
+    }
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->moriTexObjIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MORI_TEX);
+    if (this->moriTexObjIndex < 0) {
+        Actor_Kill(&this->dyna.actor);
+        // Bank danger!
+        osSyncPrintf("Error : バンク危険！(arg_data 0x%04x)(%s %d)\n", this->dyna.actor.params,
+                     "../z_bg_mori_hashira4.c", 196);
+        return;
+    }
+    if ((this->dyna.actor.params != 0) && Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    Actor_SetHeight(&this->dyna.actor, 50.0f);
+    BgMoriHashira4_SetupWaitForMoriTex(this);
+    // (4 pillars of the Forest Temple) Bank danger
+    osSyncPrintf("(森の神殿 ４本柱)(arg_data 0x%04x)\n", this->dyna.actor.params);
+    sUnkTimer = 0;
+}
+
+void BgMoriHashira4_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriHashira4* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgMoriHashira4_SetupWaitForMoriTex(BgMoriHashira4* this) {
+    BgMoriHashira4_SetupAction(this, BgMoriHashira4_WaitForMoriTex);
+}
+
+void BgMoriHashira4_WaitForMoriTex(BgMoriHashira4* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->moriTexObjIndex)) {
```

Comment:
we're gonna be focusing less on documentation so we can take care of this later if anything, but I dont agree with the function name. Its not waiting for anything here, its just the rest of the init split off into a different function

Reply:
It is waiting for the object to be loaded, and it's an action function.

Reply:
oh I see, I got the setup func mixed with the actual func and thought it was being called directly lol